mlflow/gateway/utils.py

```diff
@@ -38,3 +43,52 @@ def kill_child_processes(parent_pid):
     _, still_alive = psutil.wait_procs(parent.children(), timeout=3)
     for p in still_alive:
         p.kill()
+
+
+def _is_valid_uri(uri: str):
+    """
+    Evaluates the basic structure of a provided gateway uri to determine if the scheme and
+    netloc are provided
+    """
+    try:
+        parsed = urlparse(uri)
+        return all([parsed.scheme, parsed.netloc])
+    except ValueError:
+        return False
+
+
+def _merge_uri_paths(paths: List[str]) -> str:
+    sanitized = [part.strip("/") for part in paths]
+    merged = "/".join(sanitized)
+    if not merged.startswith("/"):
```

Comment:
Is there a case where `merged` startswith `/`?

Reply:
There was an issue in the logic in this function. I've removed it in favor of sticking with a simple urllib.parse.urljoin for get_route(). It's simpler.